src/core/update_initial_conditions.jl

```diff
@@ -82,18 +82,58 @@ function calculate_ic_quantity(
     return energy_cache
 end
 
+############################# Initial Conditions Initialization ############################
+function _make_initial_conditions!(
+    psi_container::PSIContainer,
+    devices::IS.FlattenIteratorWrapper{T},
+    key::ICKey,
+    make_ic_func::Function, # Function to make the initial condition object
+    get_val_func::Function, # Function to get the value from the device to intialize
+    cache = nothing,
+) where {T <: PSY.Component}
+    length_devices = length(devices)
+    parameters = model_has_parameters(psi_container)
+    ic_container = get_initial_conditions(psi_container)
+    if !has_initial_conditions(ic_container, key)
+        @debug "Setting $(key.ic_type) initial conditions for the status of all devices $(T) based on system data"
+        ini_conds = Vector{InitialCondition}(undef, length_devices)
+        set_initial_conditions!(ic_container, key, ini_conds)
+        for (ix, dev) in enumerate(devices)
+            val_ = get_val_func(dev, key)
+            val = parameters ? PJ.add_parameter(psi_container.JuMPmodel, val_) : val_
+            ic = make_ic_func(ic_container, dev, val, cache)
+            ini_conds[ix] = ic
+            @debug key, ic, val, 0
```

Comment:
No commas in between the variable names.  Probably don't need the 0.  I have found it to be a good practice to always include a string in the log message that you can search for. 

Also, do we want a recorder event here?

Reply:
This code runs during the simulation set-up. I would prefer that the recorder only runs to keep track of the simulation.